cmd/destroy/destroy.go

```diff
@@ -267,63 +282,320 @@ func shouldRunInRemote(opts *Options) bool {
 	return false
 }
 
-func (dc *destroyCommand) getDestroyer(ctx context.Context, opts *Options) (destroyInterface, error) {
-	var (
-		destroyer destroyInterface
-		err       error
-	)
-
+// runDestroy runs the main logic of the destroy command
+func (dc *destroyCommand) runDestroy(ctx context.Context, opts *Options) error {
+	var err error
+	isDestroyAll := false
+	isRemote := false
 	if opts.DestroyAll {
-		if !okteto.GetContext().IsOkteto {
-			return nil, oktetoErrors.ErrContextIsNotOktetoCluster
+		isDestroyAll = true
+		err = dc.destroyAll(ctx, opts)
+
+	} else {
+		// normal Destroy
+		err = dc.destroy(ctx, opts)
+
+		// Execute after the destroy function as the opts already has the manifest information to calculate it.
+		isRemote = shouldRunInRemote(opts)
+		if err == nil {
+			if opts.Name == "" {
+				oktetoLog.Success("Development environment successfully destroyed")
+			} else {
+				oktetoLog.Success("Development environment '%s' successfully destroyed", opts.Name)
+			}
 		}
-		destroyer, err = newLocalDestroyerAll(dc.k8sClientProvider, dc.executor, dc.nsDestroyer, dc.oktetoClient)
-		if err != nil {
-			return nil, err
+	}
+	metadata := &analytics.DestroyMetadata{
+		Success:      err == nil,
+		IsDestroyAll: isDestroyAll,
+		IsRemote:     isRemote,
+	}
+	dc.analyticsTracker.TrackDestroy(*metadata)
+
+	return err
+}
+
+// destroyAll executes the logic to destroy all resources within a namespace. It is different from
+// the dev environment destruction
+func (dc *destroyCommand) destroyAll(ctx context.Context, opts *Options) error {
+	if !okteto.GetContext().IsOkteto {
+		return oktetoErrors.ErrContextIsNotOktetoCluster
+	}
+	destroyer := newLocalDestroyerAll(dc.k8sClientProvider, dc.oktetoClient)
+
+	oktetoLog.Info("Destroying all...")
+
+	return destroyer.destroy(ctx, opts)
+}
+
+// destroy runs the logic needed to destroy a dev environment
+func (dc *destroyCommand) destroy(ctx context.Context, opts *Options) error {
```

Comment:
Moved some logic from `local.go` to here, as some of those things have to be executed locally are common to the local and remote operation